packages/kernel-5.15/1002-Makefile-expose-tools-builds-for-external-modules.patch

```diff
@@ -0,0 +1,61 @@
+From f39069aa728e8e7038ba0568f81b97298e47cd0f Mon Sep 17 00:00:00 2001
+From: Arnaldo Garcia Rincon <agarrcia@amazon.com>
+Date: Tue, 21 Jun 2022 16:42:09 +0000
+Subject: [PATCH 1002/1002] Makefile: expose tools/ builds for external modules
```

Comment:
As far as I can tell this is a clean revert of the upstream patch mentioned. We should name it as what it is and provide the reasoning on why we are reverting this one (as you have done in the commit message).

I will have to take a look at kmod kit to see if this is the right way to do this.

Reply:
Reverting this (https://lore.kernel.org/all/20210512065201.35268-1-masahiroy@kernel.org/) looks like a decent approach to me. The commit moved the definition of the `tools` target into a block guarded by `KBUILD_EXTMOD` (https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/Makefile?h=v5.15.43#n1117), only set when Kbuild is invoked for an out-of-tree build. I, too, would prefer a proper revert commit the way `git revert` would generate it, though.

An alternative would be to (cross-compile if needed and) package the required tools in the kmod kit, but this right here seems simpler.

Reply:
The alternative approach that I thought of was to copy the targets inside the `KBUILD_EXTMOD` block instead of reverting the commit just to make our lives easier in the future in case our revert conflicts with updates in the Makefile.

Regarding this:

> An alternative would be to (cross-compile if needed and) package the required tools in the kmod kit, but this right here seems simpler.

There is a comment in the other patch that explains why we didn't include the compiled binaries in the archive:

>Unlike other distributions, we cannot include these programs in our kernel-devel archive, because we rely on cross-compilation: these are "host" programs and may not match the architecture of the target